agent/engine/host_resource_manager.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package engine contains the core logic for managing tasks
+
+package engine
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/ecs_client/model/ecs"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
+	"github.com/aws/amazon-ecs-agent/agent/utils"
+)
+
+// HostResourceManager keeps account of each task in
+type HostResourceManager struct {
+	hostResource     map[string]*ecs.Resource
```

Comment:
Nit - maybe rename to `initialHostResource` if we plan to keep it immutable? 

Also can we add a comment for the variable to clarify that, for resources in `resourceMap`, some are "available resources" like CPU, mem, while some others are "reserved/consumed resources" like ports, so that readers don't get confused about the inconsistency.

Reply:
Thanks Yinyi, will update it!

Reply:
@Yiyuanzzz  Would it be more clearer to instead rename `hostResource` as `initialHostResource` and keep the struct name as `HostResourceManager`? As the `consumeResource` and `taskConsumed` will keep changing with state?